### PR TITLE
Add specification for VK_EXT_directfb_surface extension

### PR DESCRIPTION
**Pull request by caramelli**
I don't know what to define in the .Valid usage section for the number associated with the VUID, so this section is missing.

Nicolas Caramelli

### CODE REVIEW

**Comment by krOoze** [comment]
@caramelli You just make the Valid usage statements without the VUID. The VUIDs are generated as part of the release process. E.g. see #1269.

File: xml/vk.xml [hunk 1]
```diff
@@ -13668,5 +13668,11 @@ typedef void <name>CAMetalLayer</name>;
                 <enum value="&quot;VK_NV_extension_346&quot;"       name="VK_NV_EXTENSION_346_EXTENSION_NAME"/>
             </require>
         </extension>
+        <extension name="VK_EXT_extension_347" number="347" author="EXT" contact="Nicolas Caramelli @caramelli" supported="disabled">
```
Comment (oddhack) [inline xml/vk.xml:13671]: Having accepted the reservation PR, this is the point at which to add the actual XML description of the APIs that was in the original PR.

Comment (caramelli) [reply xml/vk.xml]: OK, I added the missing section "Valid usage" and I updated the XML description as it was in the original PR https://github.com/KhronosGroup/Vulkan-Docs/pull/1292
Modifications are available on https://github.com/caramelli/Vulkan-Docs

Nicolas Caramelli

**Comment by krOoze** [comment]
I get

> WARNING: Attempt to redefine VK_EXT_directfb_surface (this should not happen)

from building master head.

File: appendices/boilerplate.txt [hunk 2]
```diff
@@ -288,6 +288,9 @@ endif::VK_NV_win32_keyed_mutex[]
                                      | Microsoft Windows      | `vulkan_win32.h`         | `<windows.h>`                  | dname:VK_USE_PLATFORM_WIN32_KHR
 | `<<VK_KHR_xcb_surface>>`           | X11 Xcb                | `vulkan_xcb.h`           | `<xcb/xcb.h>`                  | dname:VK_USE_PLATFORM_XCB_KHR
 | `<<VK_KHR_xlib_surface>>`          | X11 Xlib               | `vulkan_xlib.h`          | `<X11/Xlib.h>`                 | dname:VK_USE_PLATFORM_XLIB_KHR
+ifdef::VK_EXT_directfb_surface[]
+| `<<VK_EXT_directfb_surface>>`      | DirectFB               | `vulkan_directfb.h`      | `<directfb.h>`                 | dname:VK_USE_PLATFORM_DIRECTFB_EXT
```
Comment (krOoze) [inline appendices/boilerplate.txt:292]: Isn't the canonical path <directfb/directfb.h>?

Comment (caramelli) [reply appendices/boilerplate.txt:292]: The warning may be fixed with https://github.com/KhronosGroup/Vulkan-Docs/pull/1302

In the Linux filesystem hierarchy standard, headers are located in /usr/include and DirectFB headers are installed in the /usr/include/directfb directory (/usr/include/directfb/directfb.h)
When compiling DirectFB applications, it is necessary to add this include directory with -I/usr/include/directfb, so #include <directfb.h> is used in a standard DirectFB code. That's why I put <directfb.h> in appendices/boilerplate.txt : is it wrong ?

Nicolas Caramelli

Comment (krOoze) [reply appendices/boilerplate.txt:292]: IDK, just inconsistent. E.g. Vulkan is installed to `/usr/include/vulkan/vulkan.h` and canonical way to include it is `<vulkan/vulkan.h>`.

Comment (caramelli) [reply appendices/boilerplate.txt:292]: OK, I made the update.

**Pull request merged**